Nautilus/Handlers/CoordinatedSpawnsHandler.cs

```diff
@@ -168,10 +169,32 @@ public SpawnInfo(TechType techType, Vector3 spawnPosition, Vector3 rotation, Vec
     /// <param name="rotation">Rotation to spawn at.</param>
     /// <param name="scale">Scale to spawn with.</param>
     public SpawnInfo(string classId, Vector3 spawnPosition, Vector3 rotation, Vector3 scale)
-        : this(classId, default, spawnPosition, Quaternion.Euler(rotation), scale) { }
+        : this(classId, default, spawnPosition, Quaternion.Euler(rotation), scale, null) { }
+    
+    /// <summary>
+    /// Initializes a new <see cref="SpawnInfo"/>.
+    /// </summary>
+    /// <param name="techType">TechType to spawn.</param>
+    /// <param name="spawnPosition">Position to spawn into.</param>
+    /// <param name="rotation">Rotation to spawn at.</param>
+    /// <param name="scale">Scale to spawn with.</param>
+    /// <param name="onSpawned">Callback that is used when the object is successfully spawned.</param>
+    public SpawnInfo(TechType techType, Vector3 spawnPosition, Quaternion rotation, Vector3 scale, Action<GameObject> onSpawned)
+        : this(default, techType, spawnPosition, rotation, scale, onSpawned) { }
 
+    /// <summary>
+    /// Initializes a new <see cref="SpawnInfo"/>.
+    /// </summary>
+    /// <param name="classId">ClassID to spawn.</param>
+    /// <param name="spawnPosition">Position to spawn into.</param>
+    /// <param name="rotation">Rotation to spawn at.</param>
+    /// <param name="scale">Scale to spawn with.</param>
+    /// <param name="onSpawned">Callback that is used when the object is successfully spawned.</param>
+    public SpawnInfo(string classId, Vector3 spawnPosition, Quaternion rotation, Vector3 scale, Action<GameObject> onSpawned)
+        : this(classId, default, spawnPosition, rotation, scale, onSpawned) { }
```

Comment:
Overall looks good to me, but don't leave this public. If we're gonna have another property that can be filled using the constructor, it will have to be added to these two, which is a breaking change.

We have the all-in-one constructor made internal for a reason.

Reply:
This comment while being directed to the PR description, the code looks perfectly fine and uses the same idea as the `SpawnInfo`'s design, so further code changes are required.